### PR TITLE
Creating a CC wavefunction (part 1 of N)

### PR DESCRIPTION
**Pull request by robertodr**
## Description

The main motivation for these changes is to make the pending #1061 available satisfying the constraints and comments offered by @dgasmith and @amjames.
This mainly amounts to localizing state to a suitable CC wavefunction object and we tried doing that within the scope of #1061, but quickly recognized that some form of rewrite of the CC infrastructure would be more helpful in general.

We are preparing a more thoroughly motivated outline of the changes we propose, which will be posted as an issue.

In the meantime, this PR makes the first step of moving the various `cc*` folders inside a top-level `cc` folder which contains the `ccwave.h` and top-level implementation files. This is thus mostly aesthetics, but avoids drowning the (hopefully!) good stuff to come in a large diff.

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] Moved `cc{energy,hbar,lambda,density,eom,response}` under `cc` folder.
- [x] Moved `ccwave.h` header file to top-level `cc` folder.
- [x] Cherry-picked some of the changes by @tomspur
   * Changes suggested by valgrind and cppcheck
   * Matrix CTOR from `dpdbuf4` @amjames could you check that setting `symmetry_` is done correctly?

Aesthetics
- [x] Run `clang-format` on `ccenergy`. Supersedes #1203, see #1144 
- [x] Run `clang-format` on `ccdensity`. Supersedes #1202, see #1144 
- [x] Run `clang-format` on `cclambda`. Supersedes #1206, see #1144 
- [x] Run `clang-format` on `cctriples`. Supersedes #1209, see #1144  
- [x] Remove `void` from function argument lists in the `cc*` folders.

## Checklist
- [ ] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: psi4/src/psi4/libmints/matrix.cc [hunk 1]
```diff
@@ -238,6 +238,29 @@ Matrix::Matrix(dpdfile2 *inFile)
     global_dpd_->file2_mat_close(inFile);
 }
 
+Matrix::Matrix(dpdbuf4 *inBuf)
+    : name_("None"), rowspi_(inBuf->params->nirreps), colspi_(inBuf->params->nirreps)
+{
+    matrix_ = NULL;
+    symmetry_ = 0; //inBuf->my_irrep;
```
Comment (amjames) [inline psi4/src/psi4/libmints/matrix.cc]: @robertodr `symmetry_ = 0` here is not correct. I'm not sure why it's commented out but `symmetry_ inBuf->my_irrep` is what we would want here.

File: psi4/src/psi4/cc/ccdensity/Frozen.h [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+/*! \defgroup CCDENSITY ccdensity: Computes the Coupled-Cluster Density */
+
+namespace psi {
+namespace ccdensity {
+
+struct Frozen {
+    int nfzc;       /* total no. of frozen core orbitals */
+    int nfzv;       /* total no. of frozen virtual orbitals */
+    int *occ_sym;   /* occupied index symmetry */
```
Comment (dgasmith) [inline psi4/src/psi4/cc/ccdensity/Frozen.h]: Would it be straightforward to move these to `std::vector`?

Comment (robertodr) [reply psi4/src/psi4/cc/ccdensity/Frozen.h]: I guess so, but that's left for later work.

**Comment by robertodr** [comment]
I did use `git mv`, so I guess the formatting is to blame. I'll start over (moving PR, then formatting PR)

**Comment by robertodr** [comment]
Closing this one, see #1253

**Pull request closed without merging**